akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala

```diff
@@ -82,7 +82,7 @@ class GraphStageLogicSpec extends StreamSpec with GraphInterpreterSpecKit {
     override def toString = "GraphStageLogicSpec.emitEmptyIterable"
   }
 
-  final case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
+  private case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
```

Comment:
The akka-stream one is tougher, it's the inner classes in `BatchingActorInputBoundary` which all access their outer scope. There unfinal them does not work and they cannot be private. One of the rabbit holes I explored today. Didn't figure out how I could solve that without changing more of it than I felt reasonable.